hack/airgapped-demo.sh

```diff
@@ -0,0 +1,262 @@
+#!/bin/bash
+set -eou pipefail
+IFS=$'\n\t'
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly SCRIPT_DIR
+
+readonly ROOT_DIR="${SCRIPT_DIR}/.."
+
+readonly DEMODATA_DIR="${ROOT_DIR}/demodata"
+rm -rf "${DEMODATA_DIR}" && mkdir -p "${DEMODATA_DIR}"
+
+export KIND_EXPERIMENTAL_DOCKER_NETWORK=shim-credentials-airgapped-kind-network
+
+docker network create --internal "${KIND_EXPERIMENTAL_DOCKER_NETWORK}" --subnet 172.255.0.0/24 --gateway 172.255.0.1 || true
+
+REGISTRY_IP=$(docker network inspect "${KIND_EXPERIMENTAL_DOCKER_NETWORK}" |
+  gojq -r '.[].IPAM.Config[].Subnet |
+                          capture("^(?<octet1and2>(?:\\d{1,3}\\.){2})(?:\\d{1,3})\\.(?:\\d{1,3})/(?:\\d{1,3})$") |
+                          .octet1and2 + "0.10"')
+
+REGISTRY_PORT=5000
+# Use a domain so it can be access on a Mac
+REGISTRY_ADDRESS=registry-airgapped
+REGISTRY_CERTS_DIR="${DEMODATA_DIR}/certs"
+REGISTRY_AUTH_DIR="${DEMODATA_DIR}/auth"
+REGISTRY_USERNAME=testuser
+REGISTRY_PASSWORD=testpassword
+
+# Create certs for the registry
+rm -rf "${REGISTRY_CERTS_DIR}"
+mkdir -p "${REGISTRY_CERTS_DIR}"
+
+OPENSSL_BIN="${OPENSSL_BIN:=openssl}"
+
+${OPENSSL_BIN} req -x509 -newkey rsa:4096 -sha256 -days 3650 -nodes \
+  -keyout "${REGISTRY_CERTS_DIR}/ca.key" -out "${REGISTRY_CERTS_DIR}/ca.crt" -subj "/CN=${REGISTRY_ADDRESS}" \
+  -addext "subjectAltName=DNS:${REGISTRY_ADDRESS},IP:${REGISTRY_IP}"
+
+sudo mkdir -p "/etc/docker/certs.d/${REGISTRY_IP}:${REGISTRY_PORT}"
+sudo cp -f "${REGISTRY_CERTS_DIR}/ca.crt" "/etc/docker/certs.d/${REGISTRY_IP}:${REGISTRY_PORT}"
+
+rm -rf "${REGISTRY_AUTH_DIR}"
+mkdir -p "${REGISTRY_AUTH_DIR}"
+docker container run --rm \
+  --entrypoint htpasswd \
+  httpd:2 -Bbn "${REGISTRY_USERNAME}" "${REGISTRY_PASSWORD}" >"${REGISTRY_AUTH_DIR}/htpasswd"
+
+docker container rm -fv registry-airgapped || true
+docker container run --rm -d \
```

Comment:
On my Mac the container doesn't get the port-mapping with the host not sure if this expected for this script with the `--internal` flag.
```
Creating cluster "airgapped-image-credential-provider-test" ...
WARNING: Overriding docker network due to KIND_EXPERIMENTAL_DOCKER_NETWORK
WARNING: Here be dragons! This is not supported currently.
 ✓ Ensuring node image (kindest/node:v1.25.3) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
ERROR: failed to create cluster: failed to get api server port: command "docker inspect --format '{{ with (index (index .NetworkSettings.Ports "6443/tcp") 0) }}{{ printf "%s	%s" .HostIp .HostPort }}{{ end }}' airgapped-image-credential-provider-test-control-plane" failed with error: exit status 1
Command Output: Template parsing error: template: :1:9: executing "" at <index (index .NetworkSettings.Ports "6443/tcp") 0>: error calling index: index of untyped nil
```

Reply:
Yeah that's "normal" and the same on Linux. It's the way that KinD tries to identify the API server port, expecting it to be exposed on a host port, yet Docker doesn't allow port mappings for internal networks.